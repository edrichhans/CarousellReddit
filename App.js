import Expo from "expo";
import React from "react";
import App from "./src/App";

export default class App1 extends React.Component {
  constructor() {
    super();
    this.state = {
      isReady: false
    };
  }

  async componentWillMount() {
    await Expo.Font.loadAsync({
      MaterialIcons: require("@expo/vector-icons/fonts/MaterialIcons.ttf"),
      // Lato: require('./assets/fonts/Lato-Regular.ttf'),
    });

    this.setState({ isReady: true });
  }
  render() {
    if (!this.state.isReady) {
      return <Expo.AppLoading />;
    }
    return <App />;
  }
}
