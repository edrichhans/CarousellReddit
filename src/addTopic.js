import React from 'react';
import {
	StyleSheet,
	Text,
	View,
	TextInput,
	TouchableOpacity
} from 'react-native';

export default class addTopic extends React.Component {
	constructor(props) {
		super(props);
		this.state = { text: 'Enter Topic here' };
	}
	render() {
		return (
			<View>
        {/* big text box */}
				<TextInput
					style={styles.textBox}
					onChangeText={(text) => this.setState({text})}
					value={this.state.text}
				/>
        {/* button */}
				<TouchableOpacity
					style={styles.button}
					onPress={() => {
            {/* use the addToList function here */}
						this.props.screenProps.addToList({score: 0, topic: this.state.text});
            {/* navigate back to home screen */}
						this.props.navigation.goBack();
					}}
				>
					<Text style={styles.buttonText}>Submit</Text>
				</TouchableOpacity>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	textBox: {
		margin: 30,
		height: 100,
	},
	button: {
		alignSelf: 'flex-end',
		margin: 20,
		borderWidth:1,
		borderColor:'rgba(0,0,0,0)',
		width:100,
		height:50,
		backgroundColor:'#ddd',
		borderRadius:0,
	},
	buttonText: {
		textAlign: 'center',
		textAlignVertical: 'center',
		alignItems: 'center'
	}
});