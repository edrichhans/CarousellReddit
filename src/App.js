import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  StatusBar
} from 'react-native';
import {
  StackNavigator,
} from 'react-navigation';
import addTopic from './addTopic.js';
import {
  Card
} from 'react-native-elements';
let Upvote = require('react-upvote');

class HomeScreen extends React.Component {
  render() {
    let topics = this.props.screenProps.topics;
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <View style={styles.listContainer}>
          <Text style={styles.header}>LIST OF TOPICS</Text>
          {
            topics.map((entry, index) => {
              return(
                <Card title={entry.topic} key={index} style={styles.card}>
                  <View style={styles.topic}>
                    <Text>{entry.topic}</Text>
                  </View>
                </Card>
              )
            })
          }
        </View>
        <TouchableOpacity
          style={styles.circleButton}
          onPress={() => navigate('addTopic')}
        >
       </TouchableOpacity>
      </View>
    );
  }
}

const SimpleApp = StackNavigator({
  Home: { 
    screen: HomeScreen,
    navigationOptions: {
      title: 'Home',
    },
  },
  addTopic: { screen: addTopic },
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  listContainer: {
    flex: 1,
  },
  circleButton: {
    alignSelf: 'flex-end',
    marginBottom: 20,
    marginRight: 20,
    borderWidth:1,
    borderColor:'rgba(0,0,0,0)',
    width:70,
    height:70,
    backgroundColor:'#f00',
    borderRadius:100,
  },
  header: {
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  card: {
    alignSelf: 'stretch',
    textAlign: 'center',
  }
});


export default class App extends React.Component{
  constructor(props) {
    super(props);
    this.state = {
      topics: []
    };
  }

  addToList(newTopic) {
    this.setState({
      topics: [...this.state.topics, newTopic]
    });
  }

  componentDidMount() {
    StatusBar.setHidden(true);
  }

  render() {
    return (
      <SimpleApp screenProps={{topics: this.state.topics, addToList: this.addToList.bind(this)}}/>
    )
  }
}
