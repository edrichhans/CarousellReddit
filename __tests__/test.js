// __tests__/Intro-test.js
import React from 'react';
import App from '../App';
import addTopic from '../src/addTopic';

import renderer from 'react-test-renderer';

it('works', () => {
    expect(1).toBe(1);
});

test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders addTopic correctly', () => {
    const tree = renderer.create(<addTopic />).toJSON();
    expect(tree).toMatchSnapshot();
});