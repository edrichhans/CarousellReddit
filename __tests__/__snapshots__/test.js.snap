// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders addTopic correctly 1`] = `<addTopic />`;

exports[`renders correctly 1`] = `<ExponentAppLoading />`;
